### PR TITLE
Fetch communities from Waku

### PR DESCRIPTION
**Pull request by prichodko**
Resolves https://github.com/status-im/community-dapp/issues/3

### CODE REVIEW

File: packages/DApp/src/hooks/useCommunities.ts [hunk 1]
```diff
@@ -23,57 +27,96 @@ export function useCommunities(publicKeys: string[]) {
...
-  }, [communitiesDetails, featureVotes, JSON.stringify(publicKeys)])
 
-  return returnCommunities
+          dispatch({
+            publicKey: deserializedPublicKey,
+            name: community!.identity!.displayName,
+            description: community!.identity!.description,
+            ens: community!.identity!.ensName,
+            icon: URL.createObjectURL(
+              new Blob([community!.identity!.images.large.payload], {
```
Comment (felicio) [inline packages/DApp/src/hooks/useCommunities.ts:73]: `images` can be undefined, unset by user

image

File: packages/DApp/src/main.tsx
Comment (felicio) [inline packages/DApp/src/main.tsx:1]: image

File: package.json [hunk 2]
```diff
@@ -13,7 +13,11 @@
     "lint:fix": "wsrun -c -s lint:fix",
     "build": "wsrun -e -c -s build",
     "test": "wsrun -e -c -s --exclude-missing test",
-    "clean": "wsrun -e -c -s clean"
+    "clean": "wsrun -e -c -s clean && rimraf node_modules"
+  },
+  "resolutions": {
+    "js-waku": "0.30.0",
+    "protons-runtime": "3.1.0"
```
Comment (felicio) [inline package.json:20]: ⚠️ ensures single occurrence and resolve issue with decoding BigInts causing skipping of messages

File: packages/DApp/package.json [hunk 3]
```diff
@@ -7,23 +7,25 @@
   "license": "MIT",
   "type": "module",
   "scripts": {
-    "build": "rm -rf dist && NODE_OPTIONS=--openssl-legacy-provider webpack --mode=production --env ENV=production",
-    "start": "NODE_OPTIONS=--openssl-legacy-provider webpack serve --mode=development --env ENV=development --https",
+    "dev": "NODE_OPTIONS=--openssl-legacy-provider webpack serve --mode=development --env ENV=localhost --https",
```
Comment (felicio) [inline packages/DApp/package.json:10]: ℹ️ set localhost

File: packages/DApp/src/constants/peers.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/**
+ * source: @see https://fleets.status.im
+ */
+
+export const peers = {
+  production: [
+    '/dns4/node-01.ac-cn-hongkong-c.status.prod.statusim.net/tcp/443/wss/p2p/16Uiu2HAkvEZgh3KLwhLwXg95e5ojM8XykJ4Kxi2T7hk22rnA7pJC',
```
Comment (felicio) [inline packages/DApp/src/constants/peers.ts:7]: 🚨 awaits https://github.com/status-im/community-dapp/issues/23#issuecomment-1532668176

File: packages/DApp/webpack.config.cjs [hunk 5]
```diff
@@ -1,90 +1,101 @@
...
+          test: /\.m?[jt]sx?$/,
+          resolve: {
+            fullySpecified: false,
             fallback: {
-                "buffer": require.resolve("buffer/"),
-                "crypto": require.resolve("crypto-browserify"),
-                "stream": require.resolve("stream-browserify"),
-                "assert": require.resolve("assert")
-            }
+              zlib: false,
```
Comment (felicio) [inline packages/DApp/webpack.config.cjs:39]: can be removed

File: packages/DApp/webpack.config.cjs [hunk 6]
```diff
@@ -1,90 +1,101 @@
...
+              cacheDirectory: true,
+              babelrc: false,
+              presets: [
+                '@babel/preset-typescript',
+                [
+                  '@babel/preset-env',
+                  {
+                    useBuiltIns: 'entry',
+                    targets: { browsers: 'chrome 77' },
+                    corejs: { version: '3.26' },
```
Comment (felicio) [inline packages/DApp/webpack.config.cjs:54]: required for desktop app

**Review by felicio — CHANGES_REQUESTED** [review]

File: packages/DApp/src/main.tsx
Comment (felicio) [inline packages/DApp/src/main.tsx:1]: Validate pk input, else

```
dex.js:10118 Uncaught Error: Invalid public key multibase code
 ```

File: packages/DApp/webpack.config.cjs [hunk 7]
```diff
@@ -1,90 +1,101 @@
...
+            options: {
+              cacheDirectory: true,
+              babelrc: false,
+              presets: [
+                '@babel/preset-typescript',
+                [
+                  '@babel/preset-env',
+                  {
+                    useBuiltIns: 'entry',
+                    targets: { browsers: 'chrome 77' },
```
Comment (felicio) [inline packages/DApp/webpack.config.cjs:53]: todo: request QtWebEngine (browser) to be updated instead, or configure target

File: packages/DApp/src/main.tsx
Comment (prichodko) [reply packages/DApp/src/main.tsx:1]: fixed in https://github.com/status-im/community-dapp/pull/24/commits/268bb741972fc625772cdec1a076c4133d221a4f

Comment (prichodko) [reply packages/DApp/src/main.tsx:1]: fixed by https://github.com/status-im/community-dapp/pull/24/commits/9d81f848d625636d1e6d80da82c309f3694d4d08#diff-a6530d60d177aa6bdfe7fdb99931ef15fb4961616d23ac8af22b9f1931d39ce8R222

[the remark this answers is not in the document]
Comment (felicio) [reply packages/DApp/src/main.tsx:1]: image

File: packages/DApp/src/components/card/CardCommunity.tsx [hunk 8]
```diff
@@ -8,8 +8,9 @@ import { VoteConfirmModal } from './VoteConfirmModal'
 import binIcon from '../../assets/images/bin.svg'
 import { RemoveModal } from './RemoveModal'
 import { CardHeading } from '../Card'
-import { useEthers } from '@usedapp/core'
+import { useEthers, getExplorerAddressLink, ChainId } from '@usedapp/core'
```
Comment (felicio) [inline packages/DApp/src/components/card/CardCommunity.tsx:11]: > can call with Chain directly, ex:
 > import { Mainnet } from '@usedapp/core'
 > Mainnet.getExplorerAddressLink('0xC7095A52C403ee3625Ce8B9ae8e2e46083b81987')
> – https://github.com/TrueFiEng/useDApp/pull/411

image

File: packages/DApp/src/hooks/useCommunities.ts [hunk 9]
```diff
@@ -23,57 +27,98 @@ export function useCommunities(publicKeys: string[]) {
...
+            description: community!.identity!.description,
+            ens: community!.identity!.ensName,
+            icon: community!.identity!.images.large
+              ? URL.createObjectURL(
+                  new Blob([community!.identity!.images.large.payload], {
+                    type: 'image/jpeg',
+                  })
+                )
+              : null,
+            link: `https://join.status.im/c/${deserializedPublicKey}}`,
```
Comment (felicio) [inline packages/DApp/src/hooks/useCommunities.ts:79]: non-blocking: seems like current site don't support this, see https://join.status.im/c/0x029f196bbfef4fa6a5eb81dd802133a63498325445ca1af1d154b1bb4542955133

File: packages/DApp/src/providers/waku/provider.tsx [hunk 10]
```diff
@@ -23,25 +23,25 @@ export function WakuProvider({ children }: Props) {
         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
         // @ts-ignore
         emitSelf: true,
```
Comment (felicio) [inline packages/DApp/src/providers/waku/provider.tsx:25]: non-blocking: try removing

**Review by felicio — APPROVED** [review]
Approved. Everything but https://github.com/status-im/community-dapp/pull/24#discussion_r1184684613, I'd say is non-blocking or not strictly related to the task and could be resolved as follow-ups.

File: packages/DApp/src/constants/peers.ts (hunk 4 above)
Comment (felicio) [reply packages/DApp/src/constants/peers.ts:7]: increased timeout at https://github.com/status-im/community-dapp/issues/23#issuecomment-1534496637

**Pull request merged**